### PR TITLE
feat: Deprecate and replace Bool,Int,Int64,String with Ptr using generics

### PR DESCRIPTION
**Pull request by alexandear**
This PR proposes adding a new helper routine `github.Ptr` that can be used instead of existing `github.Bool`, `github.Int`, `github.Int64`, `github.String`.

NOTE that this PR bumps the minimum required version of Go from 1.17 to 1.18.

### CODE REVIEW

File: github/actions_permissions_enterprise.go
Comment (gmlewis) [inline github/actions_permissions_enterprise.go:28]: I don't think this change is correct.  There may be more of these.  I'll let you find them.

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/actions_permissions_enterprise.go
Comment (alexandear) [reply github/actions_permissions_enterprise.go]: Fixed, thanks

**Comment by gmlewis** [comment]
OK, I'm reviewing now. Please refrain from force-pushing if possible, as we always squash-and-merge in this repo so the commit history will be clean, and this will make reviewing changes in the PR much easier for the reviewer(s). Thanks.

**Review by gmlewis — APPROVED** [review]
Thank you, @alexandear !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging (along with any discussion for or against this change, as usual, of course).

**Comment by alexandear** [comment]
Just for the reference: [`xanzy/go-gitlab`](https://github.com/xanzy/go-gitlab) deprecated `gitlab.String, gitlab.Bool, gitlab.Int` in favor of `gitlab.Ptr` in xanzy/go-gitlab#1788 one year ago.

**Review by stevehipwell — APPROVED** [review]
LGTM - But as a caveat I've not reviewed every line of code.

**Comment by gmlewis** [comment]
@alexandear - hearing no complaints about this PR, and having received one other LGTM (thank you, @stevehipwell !), let's go ahead and move forward with it.
If you could please merge the latest master into this PR and make sure it all still works, that would be greatly appreciated.
Thank you!

**Comment by alexandear** [comment]
> If you could please merge the latest master into this PR and make sure it all still works, that would be greatly appreciated.
Merged

**Comment by gmlewis** [comment]
Thank you, @alexandear !
Merging.

**Pull request merged**
